saga/saga.go

```diff
@@ -199,21 +205,32 @@ func (s *Saga) updateSagaState(msgs []SagaMessage) error {
 // updated in a thread safe manner
 // Processes at most SagaUpdateChSize number of updates at a time
 func (s *Saga) updateSagaStateLoop() {
+	var loopLatency = s.stat.Latency(stats.SagaUpdateStateLoopLatency_ms)
 	updates := []sagaUpdate{}
+
+	loopLatency.Time()
 	for update := range s.updateCh {
 		updates = append(updates, update)
 		if len(s.updateCh) == 0 || len(updates) == common.DefaultSagaUpdateChSize {
+			loopLatency.Stop()
+			s.stat.Gauge(stats.SagaNumUpdatesProcessed).Update(int64(len(updates)))
+
 			s.updateSaga(updates)
+
 			updates = []sagaUpdate{}
+			loopLatency.Time()
 		}
 	}
+	loopLatency.Stop()
```

Comment:
I think we do not actually want to call this as we will end up measuring an arbitrary amount of time between the last update and when s.updateCh is getting closed, which is unrelated

Reply:
yeah, makes sense